From dc5edb8fc313896210e4ba7319f4ba9ae0010ced Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Sun, 19 Oct 2014 21:05:46 +0300
Subject: [PATCH] ARM: dts: hummingboard: Setup pwm lines

Setup pwm lines as follows -
pwm1: In case HummingBoard base carrier; this pin drives through a serial
      capacitor the mono out of the audio jack.
      In case HummingBoard pro the this pad can be reached by wiring to
      C8 capacitors on the board.
pwm2: Setup pwm2 on gpio-1 but leave the default function of the iopad as
      a gpio.
      The user can change the io pad mux in user space and therefore use
      this function on gpio-1 (pin number 7 on the 26 pin header).

pwm3,pwm4: unused

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 arch/arm/boot/dts/imx6qdl-hummingboard.dtsi |   26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/arch/arm/boot/dts/imx6qdl-hummingboard.dtsi b/arch/arm/boot/dts/imx6qdl-hummingboard.dtsi
index c3ad51e..2003262 100644
--- a/arch/arm/boot/dts/imx6qdl-hummingboard.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-hummingboard.dtsi
@@ -273,6 +273,13 @@
 				MX6QDL_PAD_EIM_DA4__GPIO3_IO04 0x80000000
 			>;
 		};
+
+		pinctrl_pwm1: pwm1grp {
+			fsl,pins = <
+				MX6QDL_PAD_DISP0_DAT8__PWM1_OUT 0x1b0b1
+			>;
+		};
+
 	};
 };
 
@@ -339,3 +346,22 @@
 	no-msi;
 };
 
+&pwm1 {
+        pinctrl-names = "default";
+        pinctrl-0 = <&pinctrl_pwm1>;
+        status = "okay";
+};
+
+&pwm2 {
+        pinctrl-names = "default";
+        status = "okay";
+};
+
+&pwm3 {
+	status = "disabled";
+};
+
+&pwm4 {
+	status = "disabled";
+};
+
-- 
1.7.9.5

